spec/models/transaction_spec.rb

```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
+
+RSpec.describe Transaction, type: :model do
+  describe 'relationships' do
+    it { should belong_to(:invoice) }
+  end
+
+  describe 'validations' do
+    it { should validate_presence_of(:credit_card_number) }
+    it { should validate_presence_of(:credit_card_expiration_date) }
+    it { should validate_presence_of(:result) }
```

Comment:
Add validation test for numericality: `it { should validate_numericality_of(:attribute_name) }`